python/taichi/lang/field.py

```diff
@@ -377,6 +378,13 @@ def getter(*key):
             def setter(value, *key):
                 assert len(key) == _ti_core.get_max_num_indices()
                 snode.write_float(key, value)
+                if impl.get_runtime().target_tape and impl.get_runtime(
+                ).target_tape.grad_check and not impl.get_runtime(
+                ).grad_replaced:
+                    for x in impl.get_runtime().target_tape.grad_check:
+                        assert snode != x.snode.ptr, "You can not write to input field."
```

Comment:
I think the error message is not so informative. It might be better to mention that the reason that overwritten is prohibitive is that we are doing grad check.